### ISSUE TITLE
"Unexpected token: {" in new release on old node

### ISSUE THREAD

**Issue by Grazfather** [neoclide/coc.nvim#2798]
```javascript
[coc.nvim]: /.../dotfiles/vim/plugged/coc.nvim/build/index.js:99
`});var y=f(n.bottom,n["bottom-left"]||n.bottom,n["bottom-right"]||n.bottom,n["bottom-mid"]);return y?r+=y:r=r.slice(0,-1),r};Wx.exports=_f;Wx.exports.version=f2().version}),v2=g((nTe,d2)=>{"use strict";var h2=10,m2=(r=0)=>e=>`^[[${38+r};
5;${e}m`,g2=(r=0)=>(e,t,i)=>`^[[${38+r};2;${e};${t};${i}m`;function Cce(){let r=new Map,e={modifier:{reset:[0,0],bold:[1,22],dim:[2,22],italic:[3,23],underline:[4,24],inverse:[7,27],hidden:[8,28],strikethrough:[9,29]},color:{black:[30,39]
,red:[31,39],green:[32,39],yellow:[33,39],blue:[34,39],magenta:[35,39],cyan:[36,39],white:[37,39],blackBright:[90,39],redBright:[91,39],greenBright:[92,39],yellowBright:[93,39],blueBright:[94,39],magentaBright:[95,39],cyanBright:[96,39],w
hiteBright:[97,39]},bgColor:{bgBlack:[40,49],bgRed:[41,49],bgGreen:[42,49],bgYellow:[43,49],bgBlue:[44,49],bgMagenta:[45,49],bgCyan:[46,49],bgWhite:[47,49],bgBlackBright:[100,49],bgRedBright:[101,49],bgGreenBright:[102,49],bgYellowBright:
[103,4
SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:616:28)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
```

Did a bisect:

```
~/code/dotfiles/vim/plugged/coc.nvim] $ git bisect bad
6750be9b21a735e0e1b8ee30f4677992a52d226c is the first bad commit
commit 6750be9b21a735e0e1b8ee30f4677992a52d226c
Author: Qiming Zhao <chemzqm@gmail.com>
Date:   Fri Nov 6 16:32:48 2020 +0800

    fix(float.vim): fix bufhidden option for buffer reuse

    Closes #2556

:040000 040000 a005e7396de2fb87ac49704a4a8d473ac842acaf 01a3ab0ffe1393e28a2c8ff7801c2cf9b650cc35 M      autoload
:040000 040000 1d8d16bbb14227f900948a6cfb88b2718fdd80ed 9de3073545a8efc45a85293c21977e692d148c82 M      build
```

Note that I have old nodejs. Earlier commits (properly) complain.
```
Error: Node version v8.10.0 < 10.12.0, please upgrade node.js
```

**Comment by Grazfather** [comment]
The error I get on the first bad commit is different:

```javascript
[coc.nvim]: /home/cruise/code/dotfiles/vim/plugged/coc.nvim/build/index.js:45428
      } catch {
              ^
SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:616:28)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
```

Using neovim 0.5.0

**Comment by Grazfather** [comment]
Upgrading node DOES fix the issue, but I figure you might want to restore the ability to warn about old node, since the error we get now is useless.

**Comment by chemzqm** [comment]
> want to restore the ability to warn about old node

It's still there, but the code failed to run by your node.

**Comment by Grazfather** [comment]
Yes, the point is that you have code to report an error in your code base that is useless if it can't get that far.

**Comment by sullyj3** [comment]
I'm getting the same error as @Grazfather, even after upgrading node to `v15.11.0` with nvm. It seems to happen when I start nvim, but it's intermittent, every so often it doesn't occur.

**Comment by grillermo** [comment]
For me the error went away using rvm use 14.15.0 before opening vim.

**Comment by chemzqm** [comment]
The node version check was fixed by b427b2ccf
